forum/qa-content/qa-page.js

```diff
@@ -870,3 +876,95 @@ function qa_ajax_error()
+
+	if ( !!copyToClipboard ) {
+		var shareMethodPopup = ( function() {
+			var popup = document.createElement( 'div' );
+			popup.classList.add( 'share-popup' );
+            popup.id = 'share-popup';
+
+			var shareLinkField = document.createElement( 'input' );
+			shareLinkField.type = 'text';
+			shareLinkField.readonly = true;
+            shareLinkField.id = 'shareLinkField';
+			shareLinkField.value = '';
+
+			var copyLinkBtn = document.createElement( 'button' );
+			copyLinkBtn.type = 'button';
+            copyLinkBtn.textContent = 'Kopiuj link';
+            copyLinkBtn.classList.add( 'copy-shared-link' );
+
+			popup.appendChild( shareLinkField );
+			popup.appendChild( copyLinkBtn );
+
+			return {
+                popup: popup,
+                shareLinkField: shareLinkField
+            };
+		}() );
+
+		var addShareButton = function() {
+
+			var topicActionButtonsBar = document.querySelector( '.qa-q-view-buttons' );
+			var answersActionButtonsBar = Array.from( document.querySelectorAll( '.qa-a-item-buttons' ) );
+			var commentsActionButtonsBar = Array.from( document.querySelectorAll( '.qa-c-item-buttons' ) );
+
+			var allActionButtonsBar = answersActionButtonsBar.concat( commentsActionButtonsBar );
+			allActionButtonsBar.push( topicActionButtonsBar );
+
+			for ( var i = 0, len = allActionButtonsBar.length; i < len; i++ ) {
+				var shareButton = document.createElement( 'button' );
+                shareButton.type = 'button';
+				shareButton.textContent = 'Udostępnij';
+				shareButton.classList.add( 'share-button' );
+
+				allActionButtonsBar[ i ].appendChild( shareButton );
+			}
+
+            var popupParentRef = {};
+
+            var mainParent = document.querySelector( '.qa-main' );
+			mainParent.addEventListener( 'click', function( ev ) {
+                var eTarget = ev.target;
+                var eClassList = eTarget.classList;
+                var parentRef = eTarget.parentNode;
+
+                if ( eClassList.contains( 'share-button' ) ) {
+                    var topicContentGroup = eTarget.parentNode.classList[ 0 ].split( '-' )[ 2 ];
+                    var olderParentRef = topicContentGroup === 'item' ? parentRef.parentNode.parentNode : parentRef.parentNode;
+                    popupParentRef = parentRef;
+                    var sharePopupElem = document.getElementById( 'share-popup' );
+
+                    if ( !sharePopupElem ) {
+                        var url = olderParentRef.querySelector( '[class*="-' + topicContentGroup + '-meta"] a' ).href;
+
+                        shareMethodPopup.shareLinkField.value = url;
+                        parentRef.appendChild( shareMethodPopup.popup );
+                    } else if ( sharePopupElem ) {
+                        sharePopupElem.parentNode.removeChild( shareMethodPopup.popup );
+                    }
+				} else if ( eClassList.contains( 'copy-shared-link' ) ) {

```

Comment:
after reading the if() contents i forgot what was the conditional expression, and then there is some else if. I would suggest that this look like this:
```
if (eClassList.contains( 'share-button' )) {
    doAllStuff();
} else if (eClassList.contains( 'copy-shared-link' )) {
    doOtherStuff()
}
```
If there are some decisions trees, they should be as small as possible